packages/aws-arch/scripts/generate-assets.ts

```diff
@@ -0,0 +1,392 @@
+
+export const ASSETS_MARKDOWN = path.join(__dirname, "..", 'ASSETS.md');
+
+export const ASSET_PATTERNS = {
+  CATEGORY: /Category-Icons_\d+\/Arch-Category_64\/Arch-Category_(?<category>[^_]+)_64\.svg$/,
+  SERVICE: /Architecture-Service-Icons_\d+\/Arch_(?<category>[^/]+)\/(Arch_)?64\/Arch_(?<serviceName>[^_]+)_64\.svg$/,
+  RESOURCE: /Resource-Icons_\d+\/Res_(?<category>[^/]+)\/Res_48_(Light|Dark)\/Res_(?<serviceName>[^_]+)_(?<resourceName>[^_ ]+) ?_48_((?<theme>Light|Dark))\.svg$/,
+  GENERAL: /(Arch|Res)_General-Icons\/(64|Res_48_(Light|Dark))\/(Arch|Res)_(?<generalIcon>[^_]+)_.*(?<theme>Light|Dark).*\.svg$/,
+};
+
+const EC2_INSTANCE_PATTERN = /compute\/ec2\/(?<instanceType>\w+)_instance$/i;
+
+const IOT_THING_PATTERN = /internet_of_things\/iot_thing\/(?<thingType>\w+)$/i;
+
+export const ASSET_SIZE = 450;
+
+const DEFAULT_THEME = 'light';
+
+// List of "general" resources used by other resources from common resources
+// Lots of services have logs, rules, policies, which are pulled from IAM
+// TODO: convert generic assets to grayscale
+const GENERAL_ALIASES = {
+  alarm: 'management_governance/cloudwatch/alarm',
+  attachment: 'networking_content_delivery/transit_gateway/attachment',
+  config: 'general/gear',
+  configuration: 'general/gear',
+  credentials_long: 'security_identity_compliance/iam/long_term_security_credential',
+  credentials_temp: 'security_identity_compliance/iam/temporary_security_credential',
+  encrypted_data: 'security_identity_compliance/iam/encrypted_data',
+  event: 'management_governance/cloudwatch/event_event_based',
+  interactive_video: 'general/multimedia',
+  key: 'security_identity_compliance/iam/permissions',
+  logs: 'management_governance/cloudwatch/logs',
+  policy: 'security_identity_compliance/iam/add_on',
+  role: 'security_identity_compliance/iam/role',
+  rule: 'management_governance/cloudwatch/rule',
+  scheduled_event: 'management_governance/cloudwatch/event_time_based',
+  template: 'management_governance/cloudformation/template',
+ } as const;
+
+export async function generate () {
+  await fs.ensureDir(ASSET_PACKAGE_DIR);
+  await fs.emptyDir(ASSET_PACKAGE_DIR);
+  await fs.ensureDir(path.dirname(AWS_ASSETS_TS));
+
+
+  const assetFiles: string[] = [];
+
+  // copy static assets
+  const groups = new Set<[string, string]>();
+  for(const staticFile of await listDirFiles(path.join(STATIC_ASSETS))) {
+    if (staticFile.match(/\/groups\/\w+\.(png|svg)/)) {
+      // TODO: support svg groups - need to find the assets
+      const group = path.basename(staticFile).split('.')[0];
+      group !== "" && groups.add([group, `groups/${group}`]);
+    }
+    assetFiles.push(staticFile.replace(STATIC_ASSETS + '/', ''));
+  }
+  await fs.copy(STATIC_ASSETS, ASSET_PACKAGE_DIR, { overwrite: true, recursive: true });
+
+  const cachedir = path.join(process.cwd(), 'node_modules', '.cache', '@aws-prototyping-sdk', 'aws-arch');
+  await fs.ensureDir(cachedir);
```

Comment:
Does this still work if you blow away the cache and run generate?

Reply:
Yes - I just verified locally. Caching is just for downloaded files and deleting the cache works with subsequent generate